src/App.tsx

```diff
@@ -1,73 +1,91 @@
-import React from 'react';
+import React, { useCallback, useMemo, useRef, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { DropdownMenu } from './components/DropdawnMenu/DropdawnMenu';
+import classNames from 'classnames';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [people] = useState<Person[]>(peopleFromServer);
+  const [query, setQuery] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+  const [normalizedQuery, setNormalizedQuery] = useState('');
+  const timerId = useRef(0);
+  const field = useRef<HTMLInputElement>(null);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const checkFocus = useCallback(() => {
+    setIsFocused(field.current === document.activeElement);
+  }, []);
+
+  const saveQuery = (newQuery: string) => {
+    setQuery(newQuery);
+
+    window.clearTimeout(timerId.current);
+
+    timerId.current = window.setTimeout(() => {
+      setNormalizedQuery(newQuery.trim().toLowerCase());
+
+      if (newQuery !== normalizedQuery) {
+        setSelectedPerson(null);
+      }
```

Comment:
The condition `if (newQuery !== normalizedQuery)` might not work as expected because `setNormalizedQuery` is asynchronous. Consider using a callback or another method to ensure `normalizedQuery` is updated before this check.